pkg/envelope/jwe.go

```diff
@@ -33,15 +35,13 @@ func Encrypt(payload []byte, enc ContentEncryptionAlgorithm, keySettings []KeyEn
 }
 
 func Decrypt(in JWE, keys []any) ([]byte, error) {
-	options := make([]jwe.DecryptOption, 0, len(keys)*(len(in.Recipients)+1))
-
-	// TODO: Generate WithKey options based on the compatibility between the keys and the key encryption algorithm.
+	options := make([]jwe.DecryptOption, 0, len(keys))
 	for _, key := range keys {
-		for _, r := range in.Recipients {
-			options = append(options, jwe.WithKey(jwa.KeyEncryptionAlgorithm(r.Header.Alg), key))
-		}
-		if len(in.Recipients) == 0 {
-			options = append(options, jwe.WithKey(jwa.KeyEncryptionAlgorithm(in.Header.Alg), key))
+		switch key.(type) {
+		case *rsa.PrivateKey:
+			options = append(options, jwe.WithKey(jwa.RSA_OAEP_256, key))
+		case *ecdsa.PrivateKey:
+			options = append(options, jwe.WithKey(jwa.ECDH_ES_A256KW, key))
```

Comment:
Based on the key type to choose EncryptionAlgorithm is wrong. 
We should put the keys into options when the key is compatible with r.Header.Alg. 
For example, if r.Header.Alg is ECDH_ES, ECDH_ES_A128KW or other ECDH_ES series , the compatible key is ECDSA key.
RSA1_5, RSA_OAEP or RSA_OAEP_256 are compatible to RSA key.

Reply:
Fixed.
Create options against Recipients and Header algorithms.